clang/lib/CodeGen/CGExprScalar.cpp

```diff
@@ -2877,6 +2888,17 @@ ScalarExprEmitter::EmitScalarPrePostIncDec(const UnaryOperator *E, LValue LV,
   } else if (type->isIntegerType()) {
     QualType promotedType;
     bool canPerformLossyDemotionCheck = false;
+
+    // Is the pattern "while (i--)" and overflow exclusion?
+    bool disableSanitizer =
```

Comment:
This is difficult to read. I know it's nice to have it all in one big conditional, but it'd be more readable if split up a bit.

Reply:
You're right. I split up the logic into a new method in 4b3efbb41ff86eeff15671b1d876e1ef6a58a536